### DISCUSSION TITLE
Suggestion: Make it possible to customize FlxButton graphics without extending the class

### DISCUSSION THREAD

**Discussion by moxie-coder** [HaxeFlixel/flixel#3137] [Q&A] [answered]
There is no way to customize the graphics of the button without extending the class due to having no functions for customization, so extending the class is really the only way to get around the default graphics. I was hoping that it would be possible to customize it without even having to extend the class.

**Comment by Geokureli** [comment]
You don't need to extend it, just call `loadGraphic` on it. [example](https://github.com/HaxeFlixel/flixel-demos/blob/ba16ad7b6237067bb2b354c7144be36ef661391f/Tutorials/TurnBasedRPG/source/states/MenuState.hx#L38)

**Comment by Geokureli** [comment]
didn't mean to close this, lmk if you're still having troubles. Also, how are you customizing these by extension, if not with loadGraphic?

**Comment by moxie-coder** [comment]
> didn't mean to close this, lmk if you're still having troubles. Also, how are you customizing these by extension, if not with loadGraphic?

okay firstly, my bad since I forgot `Paths.image` which means I didn't really have to extend it, but another issue arises when trying to replace it
![image](https://github.com/HaxeFlixel/flixel/assets/124418090/14d62c11-8ecf-41a4-afa8-10c026c6266d)

here's an snippet of the code I'm using too
```haxe
			button = new FlxButton(0, 0, buttonName, function(){
				switch(buttonName){
					case 'Story mode':
						MusicBeatState.switchState(new StoryMenuState());
					case 'Freeplay':
						MusicBeatState.switchState(new FreeplayState());
					case 'Credits':
						MusicBeatState.switchState(new CreditsState());
					case 'Options':
						LoadingState.loadAndSwitchState(new options.OptionsState());
					default: null;
				}
			});
			button.loadGraphic(Paths.image('mcButton/button_medium'), true, 200, 20);
			button.animation.add('idle', [0], 24, false);
			button.animation.add('hover', [2, 1], 24, false);
			button.animation.play('idle', false);
			button.label.setFormat(Paths.font('minecraftia.ttf'), 8, FlxColor.YELLOW, "center");
			button.y += 150 + i * 100;
			button.screenCenter(X);
			add(button);
```

should we move this into a discussion?

**Reply by Geokureli** [reply]
Can you rephrase the question? I have No idea what you're asking. You obviously don't need to extend the class, and for all I know your code is the problem, here.

**Reply by moxie-coder** [reply]
Okay so, when I tried to replace the default buttons graphic with my own, an error (I’ve shown above) showed up when trying to load the graphic

**Reply by Geokureli** [reply]
What I mean is, loadGraphic works for me, if it's not working for you then maybe your path is wrong, or your Paths helper is not working right.

I don't recommend using Funkin's Paths class, I wrote that in a rush so that it would save the NG servers from loading the entire game every time, when it was getting millions of views. Instead I recommend just right clicking the asset in VSCode's file explorer and clicking "Copy Relative Path" and simply using that instead of Paths. Alternately, you could use the macro-based AssetPaths tools, [like the tutorial does](https://github.com/HaxeFlixel/flixel-demos/blob/dev/Tutorials/TurnBasedRPG/source/MenuState.hx#L38)

**Reply by moxie-coder** [reply, marked as answer]
I figured out the problem, turns out the width was wrong, sorry for wasting your time

**Reply by Geokureli** [reply]
I wonder if I can make the error log more clear, or if I can catch this issue sooner, what were the actual dimensions?

**Reply by moxie-coder** [reply]
150x20, not 200
but the original is 150x60
